api/types/network/endpoint.go

```diff
@@ -14,6 +14,7 @@ type EndpointSettings struct {
 	IPAMConfig *EndpointIPAMConfig
 	Links      []string
 	Aliases    []string
+	MacAddress string
```

Comment:
This was previously under "operational data"; will there be any ambiguity between a _user-configured_ mac-address and "no address configured,  so generated"?

Reply:
Yeah, this will be used to store both the "desired" MAC address and the actual MAC address so there will be some confusion between state and spec.

OTOH, if I add a `DesiredMacAddress` field here (and keep `MacAddress` under "operational data"), it'll show up in `container inspect` and we'll have to support this format forever.

I wish we could refactor the code to split state and spec first, but we'd have to break Go API compatibility and based on a discussion we had few weeks/months ago, I think there's no consensus on that right now.

Reply:
Also note that IPAMConfig is already affected by state / spec confusion, so the problem isn't new unfortunately -- I'll just make it slightly worse :upside_down_face:

Reply:
I think (within reason) Go compatibility is ok to break, or at least, I'm comfortable with breaking if it causes a compile-failure (so no risk at things going unnoticed). Breaking the API is something to be more careful about.

I know we discussed my draft PR for splitting the operational data; https://github.com/moby/moby/pull/46059, and that other maintainers were in favor of that change, but ISTR there was a suggestion to match "k8s" and "swarmkit" terminology (use "spec" for desired data).

Wondering now if that PR would allow us to remove the ambiguity; would _could_ mean 2 separate fields for MacAddress.